### DISCUSSION TITLE
Prisma migration not working with dolt

### DISCUSSION THREAD

**Discussion by PavelSafronov** [prisma/prisma#15786] [Q&A] [answered]
Hi Prisma folks, Pavel with Dolthub here.
Our product, dolt, provides a MySQL-like database, and one of our customers is experiencing issues while using Prisma+dolt, while the same migration works with Prisma+MySQL.

The error they are encountering is
```
Error: Error: unknown error: table not found: _prisma_migrations
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:251
```

But according to dolt DB logs, this error happens before Prisma queries the database for tables. Our current theory is that a different error is being originally thrown, and somehow causes the `table not found` error. But that's just a theory.

Here is a repro repo that can illustrate the issue: https://github.com/PavelSafronov/prisma-dolt-debug
I've included DB and Prisma logs for both the successful (MySQL) and the unsuccessful (dolt) runs.

We're kind of stuck here:
- We've enabled Prisma logs, but those don't appear to capture the engine internals. Is there a way to get engine logs?
- We've tried stepping through the Node.js code, but this in turns calls the Rust engine code, and I'm not sure how to step through that code. Is there a way to debug the Rust engine code during a migration?

Any advice you have will be really appreciated.

**Comment by nurul3101** [comment, marked as answer]
Hi @PavelSafronov 👋 
Thank you for reporting this issue, I haven't come across this kind of issue before, Is Dolt wire compatible with MySQL? 

I have created a GitHub Issue: #15788 so that our engineering team can have a look and investigate this, they should also be able to provide additional insights on what the root cause could be.

**Reply by PavelSafronov** [reply]
We're working on making dolt wire compatible with MySQL, but it sadly isn't there now, as this bug demonstrates.
Our current working theory is that Prisma is hitting one of unsupported/buggy codepaths, something that Prisma is expecting to work but dolt messes up, but then Prisma isn't handling that particular error well, which results in this misleading error.

**Reply by janpio** [reply]
We are continuing the discussion in https://github.com/prisma/prisma/issues/15788 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
